src/main/kotlin/compiler/Optimizer.kt

```diff
@@ -112,6 +112,8 @@ class Optimizer(private val coder: Coder) {
                 value(args[0].value!!)
             }
 
+            ?: consume(O_VAL, null, O_DISCARD)
+
```

Comment:
AND/OR in a statement context (like in the tests, for short-circuiting some verbcall) will compile to generating its boolean value and then discarding it.  Might as well throw that away in optimization.